app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+}
+
+func (s *VestingModuleTestSuite) TestGenesisDelayedVestingAccounts() {
+	// initial unlocked allocation to pay the gas fees
+	initBalanceForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(10))
+
+	tests := []struct {
+		name    string
+		accName string // will be filled by the code
+		endTime time.Time
+	}{
+		{
+			name:    "ValidVestingAccount",
+			endTime: time.Now().Add(time.Hour * 50),
+		},
+		{
+			name:    "ImmediateVesting",
+			endTime: time.Now(),
+		},
+		{
+			name:    "NegativeEndTime",
+			endTime: time.Now().Add(-time.Second * 10),
+		},
+	}
+
+	kr := testfactory.GenerateKeyring()
+
+	gsOpt := func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		vAccs := authtypes.GenesisAccounts{}
+		vBals := []banktypes.Balance{}
+		for i, tt := range tests {
+			tests[i].accName = fmt.Sprintf("vesting_%d", i)
+			bacc, coins := testfactory.NewBaseAccount(kr, tests[i].accName)
+			acc := vestingtypes.NewDelayedVestingAccount(bacc, coins, tt.endTime.Unix())
+			vAccs = append(vAccs, acc)
+			vBals = append(vBals, banktypes.Balance{
+				Address: acc.GetAddress().String(),
+				Coins:   acc.GetOriginalVesting().Add(initBalanceForGasFee),
+			})
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	for _, tt := range tests {
+		s.Run(tt.name, func() {
+			accAddress := getAddress(tt.accName, cctx.Keyring).String()
+
+			// Test account details correctness
+			cli := authtypes.NewQueryClient(cctx.GRPCClient)
+			res, err := cli.Account(context.Background(), &authtypes.QueryAccountRequest{
+				Address: accAddress,
+			})
+			assert.NoError(s.T(), err)
+
+			var qAcc vestingtypes.DelayedVestingAccount
+			err = qAcc.Unmarshal(res.Account.Value)
+			assert.NoError(s.T(), err)
+
+			// Checking the queried account data
+			assert.Equal(s.T(), accAddress, qAcc.GetAddress().String(), "account address must match")
+			assert.EqualValues(s.T(), tt.endTime.Unix(), qAcc.GetEndTime(), "end time must match")
+
+			assert.EqualValues(s.T(),
+				testfactory.BaseAccountDefaultBalance,
+				qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+				"original vesting must match")
+
+			/*--------*/
+
+			// Test the locking mechanism
+			// If the end time is already passed, the funds must be unlocked
+			// and we should be able to transfer some of it to another account
+			mustSucceed := tt.endTime.Before(time.Now())
+
+			assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+			/*--------*/
+
+			// Test available balance
+			balances, err := GetAccountSpendableBalance(cctx.GRPCClient, accAddress)
+			assert.NoError(s.T(), err)
+
+			expectedSpendableBal := int64(0)
+			if mustSucceed {
+				expectedSpendableBal = testfactory.BaseAccountDefaultBalance
```

Comment:
Since in delayed vesting account the locked is either all or none, here first zero amount is considered for spendable balance and then if the vesting amount is vested (unlocked), `testfactory.BaseAccountDefaultBalance` which is all vesting amount is assigned. 
yeah I agree with renaming some vars to add more clarity